src/back/glsl/mod.rs

```diff
@@ -106,6 +107,38 @@ impl Version {
     fn supports_explicit_locations(&self) -> bool {
         *self >= Version::Embedded(310) || *self >= Version::Desktop(410)
     }
+
+    /// Checks if this version supports a qualifier for the given `Interpolation`.
+    fn supports_interpolation(self, interpolation: Interpolation) -> bool {
+        use Interpolation::*;
+        match interpolation {
+            Perspective => true,                   // `smooth`: the original
+            Linear => match self {                 // `noperspective`
+                Version::Desktop(_) => true,       // added in GLSL 1.30 / OpenGL 3.0
+                Version::Embedded(_) => false,     // absent as of OpenGL ES 3.2
+            }
+            Flat => match self {                   // `flat`
```

Comment:
so it's true everywhere, right? Do we still need this `match`?

Reply:
All of this needs to use `FeatureManager` instead of this sort of ad-hoc glue.